### PR TITLE
Enable the platform specific ssd firmware upgrade during reboot

### PR DESCRIPTION
**Pull request by rkdevi27**
**- What I did**
Implemented the routines for ssd firmware upgrade via normal-reboot, fast-reboot and warm-reboot 
**- How I did it**
The SSD will be upgraded using ssd-fw-upgrade script which is being called by reboot scripts.
The files required for upgrade will be shared offline.
**- How to verify it**
The files has to be placed in /tmp and reboot(reboot/fast-reboot/warm-reboot) commands has to be issued.
Attaching the logs.

[logs.txt](https://github.com/Azure/sonic-utilities/files/4808099/logs.txt)

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

### CODE REVIEW

File: scripts/fast-reboot [hunk 1]
```diff
@@ -17,7 +17,7 @@ DEVPATH="/usr/share/sonic/device"
 PLATFORM=$(sonic-cfggen -H -v DEVICE_METADATA.localhost.platform)
 PLATFORM_PLUGIN="${REBOOT_TYPE}_plugin"
 LOG_SSD_HEALTH="/usr/bin/log_ssd_health"
-
+SSD_FW_UPDATE="/usr/bin/ssd-fw-upgrade"
```
Comment (sujinmkang) [inline scripts/fast-reboot:20]: can we change the path to the script file name as 'ssd-fw-upgrade'?

File: scripts/fast-reboot [hunk 2]
```diff
@@ -560,6 +560,11 @@ if [ -x /sbin/hwclock ]; then
     /sbin/hwclock -w || /bin/true
 fi
 
+if [ -x ${SSD_FW_UPDATE} ]; then
+    debug "updating ssd fw for${REBOOT_TYPE}"
+    ${SSD_FW_UPDATE} ${REBOOT_TYPE}
```
Comment (sujinmkang) [inline scripts/fast-reboot:565]: change this to ${DEVPATH}/${PLATFORM}/${SSD_FW_UPDATE} ${REBOOT_TYPE}

File: scripts/fast-reboot [hunk 3]
```diff
@@ -560,6 +560,11 @@ if [ -x /sbin/hwclock ]; then
     /sbin/hwclock -w || /bin/true
 fi
 
+if [ -x ${SSD_FW_UPDATE} ]; then
```
Comment (sujinmkang) [inline scripts/fast-reboot:563]: can you also change this as below ?
```
if [ -x ${DEVPATH}/${PLATFORM}/${SSD_FW_UPDATE} ]; then
```

File: scripts/reboot [hunk 4]
```diff
@@ -16,6 +16,9 @@ PLAT_REBOOT="platform_reboot"
 REBOOT_CAUSE_FILE="/host/reboot-cause/reboot-cause.txt"
 VERBOSE=no
 EXIT_NEXT_IMAGE_NOT_EXISTS=4
+SSD_FW_UPDATE="/usr/bin/ssd-fw-upgrade"
```
Comment (sujinmkang) [inline scripts/reboot:19]: same as fast-reboot

File: scripts/reboot [hunk 5]
```diff
@@ -131,6 +134,11 @@ if [ -x /sbin/hwclock ]; then
     /sbin/hwclock -w || /bin/true
 fi
 
+if [ -x ${SSD_FW_UPDATE} ]; then
```
Comment (sujinmkang) [inline scripts/reboot:137]: same as fast-reboot

File: scripts/reboot [hunk 6]
```diff
@@ -131,6 +134,11 @@ if [ -x /sbin/hwclock ]; then
...
+    debug "updating ssd fw for${REBOOT_TYPE}"
+    ${SSD_FW_UPDATE} ${REBOOT_TYPE}
```
Comment (sujinmkang) [inline scripts/reboot:139]: same as above

**Comment by rkdevi27** [comment]
Hi Sujin,

I have made the requested changes and verified it.
Raised [https://github.com/Azure/sonic-buildimage/pull/4911] for sonic-buildimage changes.

**Comment by sujinmkang** [comment]
Can you please rebase this? I will approve this once it's done!

**Comment by sujinmkang** [comment]
As Joe mentioned, please revise the PR title since we moved S6100 specific changes to build-image repo.

**Comment by rkdevi27** [comment]
Hi Sujin,

Resolved the conflicts and edited the PR title.

**Comment by rkdevi27** [comment]
retest default please

**Comment by sujinmkang** [comment]
retest this please

**Pull request merged**